src/main/java/com/indix/client/impl/IndixApiClientImpl.java

```diff
@@ -245,6 +244,7 @@ public OffersSearchResult getProductsOffersPremium(Query query) throws IndixApiE
                     });
             return searchIndixApiResponse.getResult();
         } catch (IndixApiException iae) {
+            logger.error("getProductsOffersPremium failed: " + iae.getMessage());
             throw iae;
         } catch (Exception e) {
             logger.error("getProductsOffersPremium failed: " + e.getMessage());
```

Comment:
I'm somehow still not convinced that wrapping all possible exceptions under `InternalServerException` is any useful for the user. If you guys feel otherwise, go ahead with it.